frontend/src/metabase/dashboard/components/DashboardHeader/buttons/AddFilterParameterButton/AddFilterParameterButton.tsx

```diff
@@ -6,39 +7,79 @@ import {
   hideAddParameterPopover,
   showAddParameterPopover,
 } from "metabase/dashboard/actions";
-import { ParametersPopover } from "metabase/dashboard/components/ParametersPopover";
 import { getIsAddParameterPopoverOpen } from "metabase/dashboard/selectors";
 import { useDispatch, useSelector } from "metabase/lib/redux";
-import { Popover } from "metabase/ui";
+import {
+  type ParameterSection,
+  getDashboardParameterSections,
+  getDefaultOptionForParameterSectionMap,
+} from "metabase/parameters/utils/dashboard-options";
+import { getParameterIconName } from "metabase/parameters/utils/ui";
+import { Icon, Menu, Text } from "metabase/ui";
 
 export const AddFilterParameterButton = () => {
+  const sections = getDashboardParameterSections();
   const dispatch = useDispatch();
   const isAddParameterPopoverOpen = useSelector(getIsAddParameterPopoverOpen);
+  const [maxRightSectionWidth, setMaxRightSectionWidth] = useState(0);
```

Comment:
It seems that there is no way to make it work using CSS only, so I had to use this. It triggers a subsequent re-render of this menu when opened for the first time.